scripts/create-tests.js

```diff
@@ -0,0 +1,483 @@
+  process.exit();
+}
+
+const validModes = ['reading', 'interaction'];
+const validAppliesTo = ['Screen Readers', 'JAWS', 'NVDA', 'VoiceOver', 'Orca'];
+
+const scriptDirectory = path.dirname(__filename);
+const rootDirectory = scriptDirectory.split('scripts')[0];
+const testDirectory = path.join(rootDirectory, args._[0]);
+const testDirectoryRelative = args._[0];
+
+const keysFile = path.join(rootDirectory, 'tests', 'resources', 'keys.mjs');
+
+const testsFile = path.join(testDirectory, 'data', 'tests.csv');
+const atCommandsFile = path.join(testDirectory, 'data', 'commands.csv');
+const referencesFile = path.join(testDirectory, 'data', 'references.csv');
+const javascriptDirectory = path.join(testDirectory, 'data', 'js');
+const indexFile = path.join(testDirectory,'index.html');
+
+const keyDefs = {};
+
+try {
+  fse.statSync(testDirectory);
+}
+catch (err) {
+  console.log("The test directory '" + testDirectory + "' does not exist. Check the path to tests.");
+  process.exit();
+}
+
+try {
+  fse.statSync(testsFile);
+}
+catch (err) {
+  console.log("The tests.csv file does not exist. Please create '" + testsFile + "' file.");
+  process.exit();
+}
+
+try {
+  fse.statSync(atCommandsFile);
+}
+catch (err) {
+  console.log("The at-commands.csv file does not exist. Please create '" + atCommandsFile + "' file.");
+  process.exit();
+}
+
+try {
+  fse.statSync(referencesFile);
+}
+catch (err) {
+  console.log("The references.csv file does not exist. Please create '" + referencesFile + "' file.");
+  process.exit();
+}
+
+// get Keys that are defined
+
+try {
+    // read contents of the file
+    const keys = fs.readFileSync(keysFile, 'UTF-8');
+
+    // split the contents by new line
+    const lines = keys.split(/\r?\n/);
+
+    // print all lines
+    lines.forEach((line) => {
+      let parts1 = line.split(' ');
+      let parts2 = line.split('"');
+
+      if (parts1.length > 3) {
+        let code = parts1[2].trim();
+        keyDefs[code] = parts2[1].trim();
+      }
+
+    });
+} catch (err) {
+    console.error(err);
+}
+
+// delete test files
+
+var deleteFilesFromDirectory = function(dirPath) {
+  try { var files = fs.readdirSync(dirPath); }
+  catch(e) { return; }
+  if (files.length > 0) {
+    for (var i = 0; i < files.length; i++) {
+      var filePath = dirPath + '/' + files[i];
+      if (fs.statSync(filePath).isFile()) {
+        fs.unlinkSync(filePath);
+      }
+    }
+  }
+};
+
+// Create AT commands file
+
+function createATCommandFile(cmds) {
+
+  const fname = path.join(testDirectory, 'commands.json');
+  let data = {};
+
+  function addCommand(task, mode, at, key) {
+
+    if (typeof key !== 'string' || key.length === 0) {
+      return;
+    }
+
+    if (typeof data[task] !== 'object') {
+      data[task] = {};
+    }
+
+    if (typeof data[task][mode] !== 'object') {
+      data[task][mode] = {};
+    }
+
+    if (typeof data[task][mode][at] !== 'object' ) {
```

Comment:
All of these keys should also be changed to lower case. the commands.json file should have lower case keys!